go-controller/pkg/kube/kube.go

```diff
@@ -36,6 +37,8 @@ type Kube struct {
 // SetAnnotationOnPod takes the pod object and key/value string pair to set it as an annotation
 func (k *Kube) SetAnnotationOnPod(pod *kapi.Pod, key, value string) error {
 	logrus.Infof("Setting annotations %s=%s on pod %s", key, value, pod.Name)
+	// escape double quotes in the annotation value so it can be sent as a JSON patch
+	value = strings.Replace(value, "\"", "\\\"", -1)
 	patchData := fmt.Sprintf(`{"metadata":{"annotations":{"%s":"%s"}}}`, key, value)
```

Comment:
You should probably rewrite this to use `json.Marshal`.

Reply:
@danwinship Mike Cambria is handling that in https://github.com/ovn-org/ovn-kubernetes/pull/767/files#diff-fd25950948e267a4f4efca0a2687f27eR61 and then I'll update this function to just call that one. Or something like that.